tensorboard/webapp/webapp_data_source/tb_http_client.ts

```diff
@@ -26,9 +37,20 @@ import {
 
 export {HttpErrorResponse} from '@angular/common/http';
 
+function convertFormDataToObject(formData: FormData) {
+  const result = {} as {[param: string]: string | string[]};
+  for (const [key, value] of formData.entries()) {
+    result[key] = value as string;
+  }
+  return result;
+}
+
 @Injectable()
 export class TBHttpClient implements TBHttpClientInterface {
-  constructor(private http: HttpClient) {}
+  constructor(
+    private readonly http: HttpClient,
+    private readonly store: Store<State>
```

Comment:
Should TBHttpClient be ngrx-aware? It seems to fall into the category of files that we want to keep as a pure service ignorant of ngrx.  I don't have a strong opinion about this myself but wonder if it breaks some of the team's best practices.

Reply:
Ideally, no. It invites many bad patterns so it is generally advised against but I would not mind this "hack" if it can encapsulate away the details of colabness (which we hope to soon get rid of) for now.